### PR TITLE
Makes normal borg self destruct delete the posibrain/MMI on use

### PR DESCRIPTION
**Pull request by maxion12345**
untested webedit lol
alternative to https://github.com/yogstation13/Yogstation/pull/12614
# Document the changes in your pull request

Posibrains/MMIs are deleted upon self destructing. previously, you could cheese self destruct to get around stuns and call for help while ALSO knocking down your attacker for a bit. Now, you just knock them down and can sit in dchat for a bit.

Basically, silicon combat would go something like this:
1) borg sees valid
2) valid stuns borg
3) borg self destructs, popping out the mmi/posi
4) posi/mmi screams for help now that its not stunned
5) valid dies to validhunters
6) mmi reinstalled into new borg

Now, self destruct when not emagged is now much deadly to the borg. It will most likely be an option taken when being stunned and killed, but they won't be able to use it to circumvent a stun to call for the other validhunters.

Remove this text and write all information regarding your changes that should be known and documented through the Yogstation Wiki. 
Important documentation information includes, but is not limited to: any numerical values that have been changed, any images that have to be updated, names of specific pages that will be impacted by your changes. 

# Changelog

Edit the changelog below to reflect the changes made by this PR, even if the changes are minor - required for every PR.
If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name.

:cl:  
tweak: Posibrains/MMI are destroyed on ALL self destructs. No more using to circumvent stuns and call for help.
/:cl:

### CODE REVIEW

**Comment by alexkar598** [comment]
> be subverted
> rd about to deconstruct you and ask you your laws
> self destruct
> no brain
> profit

10/10 pr, upvoted

**Comment by maxion12345** [comment]
> > be subverted
> > rd about to deconstruct you and ask you your laws
> > self destruct
> > no brain
> > profit
> 
> 10/10 pr, upvoted

"ai why did your Borg self destruct"
"Idk it just does that sometimes"

**Comment by KillerOrcaCora** [comment]
Finally...

**Comment by adamsong** [comment]
Does this affect robotics console detonations?

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -659,6 +659,8 @@
 			qdel(mmi)
 		explosion(src.loc,1,2,4,flame_range = 2)
 	else
+		if(mmi)
```
Comment (adamsong) [inline code/modules/mob/living/silicon/robot/robot.dm:662]: Move this and the one for emagged outside the emagged/syndicate if statement

**Review by adamsong — CHANGES_REQUESTED** [review]

**Comment by JamieD1** [comment]
How about you just make it so they cant self destruct when stunned.

This seems like such a stupid solution to a different problem..

**Comment by Hopekz** [comment]
If the MMI gets destroyed on a smaller explosion it would also make sense for it to get destroyed on a larger explosion.

**Comment by adamsong** [comment]
> If the MMI gets destroyed on a smaller explosion it would also make sense for it to get destroyed on a larger explosion.

Yes? The large explosion is not in question here, just the small one

**Comment by maxion12345** [comment]
closing since https://github.com/yogstation13/Yogstation/pull/12614 was merged, rendering this PR useless in the grand scheme of things

self destructing as a borg is now for all intents and purposes, useless for combat

**Pull request closed without merging**